Compile Errors ESP32

Hi,

I am trying to use your library for the esp support with an ESP32 and the AS5048A. When I compile using platformio, I get the following errors:

```
In file included from lib/AS5048A/AS5048A.cpp:3:0:
lib/AS5048A/AS5048A.h: In constructor 'AS5048A::AS5048A(byte, bool)':
lib/AS5048A/AS5048A.h:11:10: error: 'AS5048A::_cs' will be initialized after [-Werror=reorder]
  uint8_t _cs;
          ^
lib/AS5048A/AS5048A.h:9:7: error:   'bool AS5048A::errorFlag' [-Werror=reorder]
  bool errorFlag;
       ^
lib/AS5048A/AS5048A.cpp:28:1: error:   when initialized here [-Werror=reorder]
 AS5048A::AS5048A(byte cs, bool debug /*=false*/)
 ^
In file included from lib/AS5048A/AS5048A.cpp:3:0:
lib/AS5048A/AS5048A.h:12:11: error: 'AS5048A::position' will be initialized after [-Werror=reorder]
  uint16_t position;
           ^
lib/AS5048A/AS5048A.h:8:7: error:   'bool AS5048A::debug' [-Werror=reorder]
  bool debug;
       ^
lib/AS5048A/AS5048A.cpp:28:1: error:   when initialized here [-Werror=reorder]
 AS5048A::AS5048A(byte cs, bool debug /*=false*/)
 ^
cc1plus: some warnings being treated as errors
*** [.pio/build/featheresp32/lib360/AS5048A/AS5048A.cpp.o] Error 1
```

Any idea what the issue is here?

Thanks.

--- a/src/AS5048A.cpp
+++ b/src/AS5048A.cpp
@@ -2,35 +2,31 @@
 
 #include <AS5048A.h>
 
-static const uint16_t AS5048A_CLEAR_ERROR_FLAG              = 0x0001;
-static const uint16_t AS5048A_PROGRAMMING_CONTROL           = 0x0003;
-static const uint16_t AS5048A_OTP_REGISTER_ZERO_POS_HIGH    = 0x0016;
-static const uint16_t AS5048A_OTP_REGISTER_ZERO_POS_LOW     = 0x0017;
-static const uint16_t AS5048A_DIAG_AGC                      = 0x3FFD;
-static const uint16_t AS5048A_MAGNITUDE                     = 0x3FFE;
-static const uint16_t AS5048A_ANGLE                         = 0x3FFF;
+static const uint16_t AS5048A_CLEAR_ERROR_FLAG = 0x0001;
+static const uint16_t AS5048A_PROGRAMMING_CONTROL = 0x0003;
+static const uint16_t AS5048A_OTP_REGISTER_ZERO_POS_HIGH = 0x0016;
+static const uint16_t AS5048A_OTP_REGISTER_ZERO_POS_LOW = 0x0017;
+static const uint16_t AS5048A_DIAG_AGC = 0x3FFD;
+static const uint16_t AS5048A_MAGNITUDE = 0x3FFE;
+static const uint16_t AS5048A_ANGLE = 0x3FFF;
 
-static const uint8_t AS5048A_AGC_FLAG												= 0xFF;
-static const uint8_t AS5048A_ERROR_PARITY_FLAG							= 0x04;
-static const uint8_t AS5048A_ERROR_COMMAND_INVALID_FLAG			= 0x02;
-static const uint8_t AS5048A_ERROR_FRAMING_FLAG							= 0x01;
+static const uint8_t AS5048A_AGC_FLAG = 0xFF;
+static const uint8_t AS5048A_ERROR_PARITY_FLAG = 0x04;
+static const uint8_t AS5048A_ERROR_COMMAND_INVALID_FLAG = 0x02;
+static const uint8_t AS5048A_ERROR_FRAMING_FLAG = 0x01;
 
-static const uint16_t AS5048A_DIAG_COMP_HIGH								= 0x2000;
-static const uint16_t AS5048A_DIAG_COMP_LOW									= 0x1000;
-static const uint16_t AS5048A_DIAG_COF											= 0x0800;
-static const uint16_t AS5048A_DIAG_OCF											= 0x0400;
+static const uint16_t AS5048A_DIAG_COMP_HIGH = 0x2000;
+static const uint16_t AS5048A_DIAG_COMP_LOW = 0x1000;
+static const uint16_t AS5048A_DIAG_COF = 0x0800;
+static const uint16_t AS5048A_DIAG_OCF = 0x0400;
 
-static const double 	  AS5048A_MAX_VALUE 					= 8191.0;
+static const double AS5048A_MAX_VALUE = 8191.0;
 
 /**
  * Constructor
  */
 AS5048A::AS5048A(byte cs, bool debug /*=false*/)
-: _cs(cs)
-, errorFlag(false)
-, ocfFlag(false)
-, position(0)
-, debug(debug)
+	: _cs(cs), errorFlag(false), ocfFlag(false), position(0), debug(debug)
 {
 }
 
@@ -38,7 +34,8 @@ AS5048A::AS5048A(byte cs, bool debug /*=false*/)
  * Initialiser
  * Sets up the SPI interface
  */
-void AS5048A::begin(){
+void AS5048A::begin()
+{
 	setDelay();
 
 	// 1MHz clock (AMS should be able to accept up to 10MHz)
@@ -55,14 +52,16 @@ void AS5048A::begin(){
  * Closes the SPI connection
  * SPI has an internal SPI-device counter, for each begin()-call the close() function must be called exactly 1 time
  */
-void AS5048A::close(){
+void AS5048A::close()
+{
 	SPI.end();
 }
 
 /**
  * Utility function used to calculate even parity of an unigned 16 bit integer
  */
-uint8_t AS5048A::spiCalcEvenParity(uint16_t value){
+uint8_t AS5048A::spiCalcEvenParity(uint16_t value)
+{
 	uint8_t cnt = 0;
 
 	for (uint8_t i = 0; i < 16; i++)
@@ -76,20 +75,20 @@ uint8_t AS5048A::spiCalcEvenParity(uint16_t value){
 	return cnt & 0x1;
 }
 
-
-
 /**
  * Get the rotation of the sensor relative to the zero position.
  *
  * @return {int16_t} between -2^13 and 2^13
  */
-int16_t AS5048A::getRotation(){
+int16_t AS5048A::getRotation()
+{
 	uint16_t data;
 	int16_t rotation;
 
 	data = AS5048A::getRawRotation();
 	rotation = static_cast<int16_t>(data) - static_cast<int16_t>(this->position);
-	if(rotation > AS5048A_MAX_VALUE) rotation = -((0x3FFF)-rotation); //more than -180
+	if (rotation > AS5048A_MAX_VALUE)
+		rotation = -((0x3FFF) - rotation); //more than -180
 
 	return rotation;
 }
@@ -97,7 +96,8 @@ int16_t AS5048A::getRotation(){
 /**
  * Returns the raw angle directly from the sensor
  */
-int16_t AS5048A::getRawRotation(){
+int16_t AS5048A::getRawRotation()
+{
 	return AS5048A::read(AS5048A_ANGLE);
 }
 
@@ -107,7 +107,8 @@ int16_t AS5048A::getRawRotation(){
   * @return {double} between 0 and 360
   */
 
-double AS5048A::getRotationInDegrees(){
+double AS5048A::getRotationInDegrees()
+{
 	int16_t rotation = getRotation();
 	double degrees = 360.0 * (rotation + AS5048A_MAX_VALUE) / (AS5048A_MAX_VALUE * 2.0);
 	return degrees;
@@ -119,7 +120,8 @@ double AS5048A::getRotationInDegrees(){
   * @return {double} between 0 and 2 * PI
   */
 
-double AS5048A::getRotationInRadians(){
+double AS5048A::getRotationInRadians()
+{
 	int16_t rotation = getRotation();
 	double radians = PI * (rotation + AS5048A_MAX_VALUE) / AS5048A_MAX_VALUE;
 	return radians;
@@ -129,18 +131,21 @@ double AS5048A::getRotationInRadians(){
  * returns the value of the state register
  * @return unsigned 16 bit integer containing flags
  */
-uint16_t AS5048A::getState(){
+uint16_t AS5048A::getState()
+{
 	return AS5048A::read(AS5048A_DIAG_AGC);
 }
 
 /**
  * Print the diagnostic register of the sensor
  */
-void AS5048A::printState(){
+void AS5048A::printState()
+{
 	if (this->debug)
 	{
 		uint16_t data = AS5048A::getState();
-		if(AS5048A::error()){
+		if (AS5048A::error())
+		{
 			Serial.print("Error bit was set!");
 		}
 		Serial.println(data, BIN);
@@ -151,7 +156,8 @@ void AS5048A::printState(){
  * Returns the value used for Automatic Gain Control (Part of diagnostic
  * register)
  */
-uint8_t AS5048A::getGain(){
+uint8_t AS5048A::getGain()
+{
 	uint16_t data = AS5048A::getState();
 	return static_cast<uint8_t>(data & AS5048A_AGC_FLAG);
 }
@@ -159,7 +165,8 @@ uint8_t AS5048A::getGain(){
 /**
  * Get diagnostic
  */
-String AS5048A::getDiagnostic(){
+String AS5048A::getDiagnostic()
+{
 	uint16_t data = AS5048A::getState();
 	if (data & AS5048A_DIAG_COMP_HIGH)
 	{
@@ -184,15 +191,19 @@ String AS5048A::getDiagnostic(){
 /*
  * Get and clear the error register by reading it
  */
-String AS5048A::getErrors(){
+String AS5048A::getErrors()
+{
 	uint16_t error = AS5048A::read(AS5048A_CLEAR_ERROR_FLAG);
-	if (error & AS5048A_ERROR_PARITY_FLAG){
+	if (error & AS5048A_ERROR_PARITY_FLAG)
+	{
 		return "Parity Error";
 	}
-	if (error & AS5048A_ERROR_COMMAND_INVALID_FLAG){
+	if (error & AS5048A_ERROR_COMMAND_INVALID_FLAG)
+	{
 		return "Command invalid";
 	}
-	if (error & AS5048A_ERROR_FRAMING_FLAG){
+	if (error & AS5048A_ERROR_FRAMING_FLAG)
+	{
 		return "Framing error";
 	}
 	return "";
@@ -201,21 +212,24 @@ String AS5048A::getErrors(){
 /*
  * Set the zero position
  */
-void AS5048A::setZeroPosition(uint16_t position){
+void AS5048A::setZeroPosition(uint16_t position)
+{
 	this->position = position % 0x3FFF;
 }
 
 /*
  * Returns the current zero position
  */
-uint16_t AS5048A::getZeroPosition(){
+uint16_t AS5048A::getZeroPosition()
+{
 	return this->position;
 }
 
 /*
  * Check if an error has been encountered.
  */
-bool AS5048A::error(){
+bool AS5048A::error()
+{
 	return this->errorFlag;
 }
 
@@ -224,12 +238,13 @@ bool AS5048A::error(){
  * Takes the address of the register as an unsigned 16 bit
  * Returns the value of the register
  */
-uint16_t AS5048A::read(uint16_t registerAddress){
+uint16_t AS5048A::read(uint16_t registerAddress)
+{
 	uint16_t command = 0x4000; // PAR=0 R/W=R
 	command = command | registerAddress;
 
 	//Add a parity bit on the the MSB
-	command |= static_cast<uint16_t>(spiCalcEvenParity(command)<<0xF);
+	command |= static_cast<uint16_t>(spiCalcEvenParity(command) << 0xF);
 
 	if (this->debug)
 	{
@@ -245,7 +260,7 @@ uint16_t AS5048A::read(uint16_t registerAddress){
 	//Send the command
 	digitalWrite(this->_cs, LOW);
 	SPI.transfer16(command);
-	digitalWrite(this->_cs,HIGH);
+	digitalWrite(this->_cs, HIGH);
 
 	delay(this->esp32_delay);
 
@@ -264,21 +279,22 @@ uint16_t AS5048A::read(uint16_t registerAddress){
 	}
 
 	//Check if the error bit is set
-	if (response & 0x4000) {
+	if (response & 0x4000)
+	{
 		if (this->debug)
 		{
 			Serial.println("Setting error bit");
 		}
 		this->errorFlag = true;
 	}
-	else {
+	else
+	{
 		this->errorFlag = false;
 	}
 
 	//Return the data, stripping the parity and error bits
 	return response & ~0xC000;
 }
-
 
 /*
  * Write to a register
@@ -287,13 +303,14 @@ uint16_t AS5048A::read(uint16_t registerAddress){
  * Returns the value of the register after the write has been performed. This
  * is read back from the sensor to ensure a sucessful write.
  */
-uint16_t AS5048A::write(uint16_t registerAddress, uint16_t data) {
+uint16_t AS5048A::write(uint16_t registerAddress, uint16_t data)
+{
 
 	uint16_t command = 0x0000; // PAR=0 R/W=W
 	command |= registerAddress;
 
 	//Add a parity bit on the the MSB
-	command |= static_cast<uint16_t>(spiCalcEvenParity(command)<<0xF);
+	command |= static_cast<uint16_t>(spiCalcEvenParity(command) << 0xF);
 
 	if (this->debug)
 	{
@@ -309,13 +326,13 @@ uint16_t AS5048A::write(uint16_t registerAddress, uint16_t data) {
 	//Start the write command with the target address
 	digitalWrite(this->_cs, LOW);
 	SPI.transfer16(command);
-	digitalWrite(this->_cs,HIGH);
+	digitalWrite(this->_cs, HIGH);
 
 	uint16_t dataToSend = 0x0000;
 	dataToSend |= data;
 
 	//Craft another packet including the data and parity
-	dataToSend |= static_cast<uint16_t>(spiCalcEvenParity(dataToSend)<<0xF);
+	dataToSend |= static_cast<uint16_t>(spiCalcEvenParity(dataToSend) << 0xF);
 
 	if (this->debug)
 	{
@@ -324,9 +341,9 @@ uint16_t AS5048A::write(uint16_t registerAddress, uint16_t data) {
 	}
 
 	//Now send the data packet
-	digitalWrite(this->_cs,LOW);
+	digitalWrite(this->_cs, LOW);
 	SPI.transfer16(dataToSend);
-	digitalWrite(this->_cs,HIGH);
+	digitalWrite(this->_cs, HIGH);
 
 	delay(this->esp32_delay);
 

--- a/src/AS5048A.h
+++ b/src/AS5048A.h
@@ -3,13 +3,14 @@
 
 #include <SPI.h>
 
-class AS5048A{
+class AS5048A
+{
 private:
-	bool debug;
+	uint8_t _cs;
 	bool errorFlag;
 	bool ocfFlag; // Avoid printing OCF flag everytime
-	uint8_t _cs;
 	uint16_t position;
+	bool debug;
 	uint8_t esp32_delay;
 
 	SPISettings settings;
@@ -55,11 +56,10 @@ private:
 	bool error();
 
 public:
-
 	/**
 	 *	Constructor
 	 */
-	AS5048A(uint8_t arg_cs, bool debug=false);
+	AS5048A(uint8_t arg_cs, bool debug = false);
 
 	/**
 	 * Initialiser

